aten/src/ATen/core/ivalue_inl.h

```diff
@@ -157,53 +101,35 @@ struct CAFFE2_API ConstantString final : c10::intrusive_ptr_target {
       const ConstantString& v);
 };
 
-template <typename Elem>
-struct CAFFE2_API List : c10::intrusive_ptr_target {
- private:
-  c10::ListPtr<Elem> elements_;
+struct Future;
 
- public:
-  typedef Elem ElemType;
+struct CAFFE2_API TuplePtr final {
```

Comment:
minor: I wonder whether we can make it to be at the same level as ListPtr instead of being a wrapper. It feel weird that tuples have .elements() but lists and dicts don't.  Can we just inherit (maybe private inheritance?) from ListPtr and expose only non-mutating methods?

Reply:
Composition > Inheritance. I could add the ListPtr API as forwarding functions to Tuple if we want to do that. Don't think it's high pri though.